site/src/pages/docs/server-annotated-service.mdx

```diff
@@ -1373,5 +1373,80 @@ public class MyAnnotatedService {
     public HttpResponse update() { ... }
 }
 ```
+
+## Specifying the default service name
+
+The default name of an annotated service which served a request is the innermost class name.
+The service name is often used as a meter tag or distributed trace's span name.
+If you want to customize the default service name, then you can override the name by annotating
+a class or method with <type://@ServiceName> like the below:
+
+```java
+public class MyService {
+    @Get("/")
+    public String get(ServiceRequestContext ctx) {
+        assert ctx.log().ensureAvailable(RequestLogProperty.NAME).serviceName()
+                  .equals(MyService.class.getName());
+    }
+}
+
+// Override the default service name by the class annotation
+@ServiceName("my-service")
+public class MyService {
+    @Get("/")
+    public String get(ServiceRequestContext ctx) {
+        assert ctx.log().ensureAvailable(RequestLogProperty.NAME).serviceName()
+                  .equals("my-service");
+    }
+
+    // Override the default service name by the method annotation
+    @ServiceName("my-post-service")
+    @Post("/")
+    public String post(ServiceRequestContext ctx) {
+        assert ctx.log().ensureAvailable(RequestLogProperty.NAME).serviceName()
+                  .equals("my-post-service");
+    }
+}
+
+// Or the default name could get overridden programmatically using a decorator.
+ServerBuilder sb = Server.builder();
+sb.annotatedService("/", new MyService(), service -> {
+    return service.decorate((delegate, ctx, req) -> {
+        ctx.logBuilder().name("my-decorated-service", ctx.method().name());
+        return delegate.serve(ctx, req);
+    });
+});
+```
+
+<Tip>
+
+Note that <type://RequestOnlyLog#name()> is safely accessed using
+`ctx.log().ensureAvailable(RequestLogProperty.NAME)` only in your annotated service method.
+Generally, it is recommended to use <type://RequestLogAccess#whenComplete()>.
+Please see [Availability of properties](https://armeria.dev/docs/advanced-structured-logging#availability-of-properties)
+before using it.
+
+If you access <type://RequestOnlyLog#name()> in a decorator or <type://HttpService>, you should use
+<type://RequestLogAccess#whenComplete()> or <type://RequestLogAccess#whenAvailable(RequestLogProperty)>
+with <type://RequestLogProperty#NAME>.
+
+```java
+ServiceRequestContext ctx = ...
+// Simply wait until a log is completed
+ctx.log().whenComplete().thenApply(log -> {
+    String serviceName = log.serviceName();
+    ...
+})
+// If you are only interested in RequestOnlyLog.name()
+ctx.log().whenAvailable(RequestLogProperty.NAME).thenApply(log -> {
+    String serviceName = log.serviceName();
+    ...
+})
+```
+
+</Tip>
```

Comment:
I think this is too much to be a tip. How about moving/merging it to `advanced-structured-logging`?